### PR TITLE
Add keyboard shortcuts

### PR DESCRIPTION
**Pull request by archdragon**
## Add keyboard shortcuts

Adds keyboard shortcuts - as mentioned in: https://github.com/elixir-lang/ex_doc/issues/967

image

## Included shortcuts:

* `s` - go to search - focuses the search bar
* `n` - toggle night mode
* `c` - toggle sidebar

Also includes a modal with shortcut info

* `?` (`shift` + `/`) - opens help modal

## Extra notes

Shortcuts will not trigger if you are typing inside inuput/textarea.

If the sidebar is closed `s` will open it and focus the search input.

### CODE REVIEW

File: assets/js/templates/keyboard-shortcuts-help-modal.handlebars [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      <div class="modal-header">
+        <div class="modal-title">Keyboard Shortcuts</div>
+        <div class="modal-close">×</div>
+      </div>
+      <div class="modal-body">
+        {{#each shortcuts}}
+          <dl class="shortcut-row">
+            <dt class="shortcut-keys">
+              {{#if this.requiresShiftKey}}
+                <div class="shortcut-key">shift</div>
```
Comment (wojtekmach) [inline assets/js/templates/keyboard-shortcuts-help-modal.handlebars]: what do you think about using `<kbd>shift</kbd>` etc for keys, like GitHub does?

File: assets/js/keyboard-shortcuts.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+// Public Methods
+// --------------
+
+export function initialize () {
+  const helpModal = helpModalTemplate({shortcuts: keyboardShortcuts})
+  $('body').append(helpModal)
+
+  $(helpModalSelector).on('keydown', function (e) {
+    if (e.keyCode === 27) { // escape key
```
Comment (wojtekmach) [inline assets/js/keyboard-shortcuts.js]: what do you think about using escape key to also de-focus search bar?

Comment (archdragon) [reply assets/js/keyboard-shortcuts.js]: 👍 Makes sense - just checked and Google search, github and google are all handling the ESC this way. I'll add it.

**Comment by josevalim** [comment]
Beautiful! ❤️

File: assets/js/templates/keyboard-shortcuts-help-modal.handlebars [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      <div class="modal-header">
+        <div class="modal-title">Keyboard Shortcuts</div>
+        <div class="modal-close">×</div>
+      </div>
+      <div class="modal-body">
+        {{#each shortcuts}}
+          <dl class="shortcut-row">
+            <dt class="shortcut-keys">
+              {{#if this.requiresShiftKey}}
+                <kbd class="shortcut-key">shift</kbd>
```
Comment (josevalim) [inline assets/js/templates/keyboard-shortcuts-help-modal.handlebars]: I don't know anything about CSS and HTML best practices but I do have a quick question: do we need the class now that we are using a very specific `kbd` tag? Would it be better to style the tag instead?

Comment (archdragon) [reply assets/js/templates/keyboard-shortcuts-help-modal.handlebars]: 👍 Added the styles directly to the KBd

**Comment by archdragon** [comment]
> Beautiful! ❤️

Thanks! 

> what do you think about using escape key to also de-focus search bar?

done!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**